forge-gui/res/cardsfolder/g/glissas_retriever.txt

```diff
@@ -5,7 +5,7 @@ PT:6/6
 K:Haste
 K:Toxic:3
 S:Mode$ CantBlockBy | ValidAttacker$ Card.Self | ValidBlocker$ Creature.powerLE2 | Description$ CARDNAME can't be blocked by creatures with power 2 or less.
-T:Mode$ ChangesZone | Origin$ Any | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ Corrupted — When CARDNAME dies, exile it. When you do, return up to X target cards from your graveyard to your hand, where X is the number of opponents who have three or more poison counters.
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ Corrupted — When CARDNAME dies, exile it. When you do, return up to X target cards from your graveyard to your hand, where X is the number of opponents who have three or more poison counters.
 SVar:TrigExile:DB$ ChangeZone | Hidden$ True | Origin$ All | Destination$ Exile | Defined$ TriggeredCard | SubAbility$ DBImmediateTrigger
 SVar:DBImmediateTrigger:DB$ ImmediateTrigger | Execute$ TrigChangeZone | TriggerDescription$ When you do, return up to X target cards from your graveyard to your hand, where X is the number of opponents who have three or more poison counters.
```

Comment:
The Exile part doesn't has a "When you do" connection

if for example a token copy of this dies, it would still trigger, but you shouldn't be able to exile it (because its a token) so "when you do" should not trigger

for this the Exile should check Remember?

Reply:
`Name:Glissa's Retriever
ManaCost:5 G
Types:Creature Phyrexian Beast
PT:6/6
K:Haste
K:Toxic:3
S:Mode$ CantBlockBy | ValidAttacker$ Card.Self | ValidBlocker$ Creature.powerLE2 | Description$ CARDNAME can't be blocked by creatures with power 2 or less.
T:Mode$ ChangesZone | Origin$ Any | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ Corrupted — When CARDNAME dies, exile it. When you do, return up to X target cards from your graveyard to your hand, where X is the number of opponents who have three or more poison counters.
SVar:TrigExile:DB$ ChangeZone | Hidden$ True | Origin$ Battlefield | Destination$ Exile | Defined$ TriggeredCard | RememberChanged$ True | SubAbility$ DBImmediateTrigger
SVar:DBImmediateTrigger:DB$ ImmediateTrigger | ConditionDefined$ Remembered | ConditionPresent$ Card.Self | Execute$ TrigChangeZone | TriggerDescription$ When you do, return up to X target cards from your graveyard to your hand, where X is the number of opponents who have three or more poison counters.
SVar:TrigChangeZone:DB$ ChangeZone | TargetMin$ 0 | TargetMax$ X | ValidTgts$ Card.YouOwn | TgtPrompt$ Select X target cards in your graveyard | Origin$ Graveyard | Destination$ Hand | SubAbility$ DBCleanup
SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
DeckHas:Ability$Graveyard
DeckHints:Ability$Mill|Dredge
SVar:X:PlayerCountOpponents$HasPropertyIsCorrupted
Oracle:Haste, toxic 3 (Players dealt combat damage by this creature also get three poison counters.)\nGlissa's Retriever can't be blocked by creatures with power 2 or less.\nCorrupted — When Glissa's Retriever dies, exile it. When you do, return up to X target cards from your graveyard to your hand, where X is the number of opponents who have three or more poison counters.`

This is my current attempt at scripting that, but upon testing, token copies are still triggering DBImmediateTrigger. I'm unsure why this is the case nor how to fix that, so I'm posting it here in the hopes that someone with better knowledge of card scripting can help.

Reply:
`ConditionDefined$ Self` + `SubAbility$ DBCleanup` needs to be on the trigger line above

Reply:
Multiple problems from the scripter on the Exile Effect:
- Remove `Hidden$ True`
- `Origin$ Graveyard`
- Use `Defined$ TriggeredNewCardLKICopy`